R/otu_table.R

```diff
@@ -21,15 +24,24 @@ read_qiime_otu_table <- function(filepath, commented=TRUE, metadata=TRUE) {
   }
   col_names <- strsplit(header_line, "\t")[[1]]
 
-  col_classes <- rep("numeric", times=length(col_names))
-  col_classes[1] <- "character"
-  if (metadata) {
-    col_classes[length(col_classes)] <- "character"
+  if (use.readr & requireNamespace("readr")) {
+    
+    full_otu_table <- as.data.frame(readr::read_tsv(
+      filepath, col_names = col_names, skip = 2))
```

Comment:
I think `skip` should be 1 if `commented` is `FALSE`, and should be 2 if `commented` is `TRUE`.